plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/AssetPackStateFromDictionary.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  	Copyright 2020 Google LLC
+ *  	Licensed under the Apache License, Version 2.0 (the "License");
+ *  	you may not use this file except in compliance with the License.
+ *  	You may obtain a copy of the License at
+ *  		https://www.apache.org/licenses/LICENSE-2.0
+ *  	Unless required by applicable law or agreed to in writing, software
+ *  	distributed under the License is distributed on an "AS IS" BASIS,
+ *  	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  	See the License for the specific language governing permissions and
+ *  	limitations under the License.
+ */
+
+package com.google.play.core.godot.assetpacks.utils;
+
+import com.google.android.play.core.assetpacks.AssetPackState;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.godotengine.godot.Dictionary;
+
+/**
+ * This class extends the AssetPackState abstract class, and provides constructor that allows
+ * PlayAssetDeliveryUtils class to instantiate AssetPackStateFromDictionary given a Godot
+ * Dictionary.
+ */
+public class AssetPackStateFromDictionary extends AssetPackState {
+  private String name;
+  private int status;
+  private int errorCode;
+  private long bytesDownloaded;
+  private long totalBytesToDownload;
+  private int transferProgressPercentage;
+
+  private static final Set<String> dictionaryRequiredKeySet =
+      new HashSet<>(
+          Arrays.asList(
+              "name",
+              "status",
+              "errorCode",
+              "bytesDownloaded",
+              "totalBytesToDownload",
+              "transferProgressPercentage"));
+
+  public AssetPackStateFromDictionary(Dictionary dict) throws IllegalArgumentException {
+    if (dict.keySet().containsAll(dictionaryRequiredKeySet)) {
+      try {
+        this.name = (String) dict.get("name");
+        this.status = (int) dict.get("status");
+        this.errorCode = (int) dict.get("errorCode");
+        this.bytesDownloaded = (long) dict.get("bytesDownloaded");
+        this.totalBytesToDownload = (long) dict.get("totalBytesToDownload");
+        this.transferProgressPercentage = (int) dict.get("transferProgressPercentage");
+      } catch (Exception e) {
```

Comment:
Nice catch! I will add this catch.

Reply:
Pun unintended? :P

You can also remove the other catch since any uncaught exception will bubble up to the caller automatically. No need to re-throw the exception.

Reply:
😁 Removed the redundant try throw clause